mailpoet/assets/js/src/automation/components/option-button/index.tsx

```diff
@@ -0,0 +1,81 @@
+import { ReactNode, useEffect, useRef, useState } from 'react';
+import { Button } from '@wordpress/components';
+import { chevronDown, chevronUp } from '@wordpress/icons';
+import { ReactNodeArray } from 'prop-types';
+
+type OptionButtonPropType = {
+  variant: Button.ButtonVariant;
+  children: ReactNodeArray | ReactNode;
+  title: string;
+  onClick: () => void;
+};
+export function OptionButton({
+  children,
+  title,
+  onClick,
+  variant,
+}: OptionButtonPropType): JSX.Element {
+  const [isOpen, setIsOpen] = useState(false);
+  const ref = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const element = ref?.current;
+
+    // Any mouse click outside the element will close the options.
+    function handleCloseOnClick(event: Event) {
+      if (element && !element.contains(event.target as Node | null)) {
+        setIsOpen(false);
+      }
+    }
+
+    // Pressing escape will close the options.
+    function handleCloseOnEsc(event: KeyboardEvent) {
+      if (event.key === 'Escape') {
+        setIsOpen(false);
+      }
+    }
+    document.addEventListener('mousedown', handleCloseOnClick, {
+      capture: true,
+    });
+    document.addEventListener('keyup', handleCloseOnEsc, { capture: true });
```

Comment:
Why did you choose not to use `DropdownMenu` and implement this manually instead? If we have no strong reason to do our own solution, I'd prefer to use the dropdown menu as it will correctly handle all of the keyboard and accessibility features under all circumstances, OSs, and browsers, for us.

Reply:
The closest to the Figma I cam using the `DropDownMenu` was the following:
![image](https://user-images.githubusercontent.com/6458412/199412548-da653ac0-d2ee-4230-b79b-ba5408f1eb3f.png)

```
    <div className="mailpoet-option-button" ref={ref}>
        <Button
          variant={variant}
          className="mailpoet-option-button-main"
          onClick={onClick}
        >
          {title}
        </Button>
      <DropdownMenu
        className="is-primary"
        label={'...'}
        icon={chevronDown}
        popoverProps={{className:'test',getAnchorRect:() => {
            const element:Element = ref.current;
            return element.getBoundingClientRect();
          }}}
        controls={[{title:'test', icon:chevronUp, isDisabled:false, onClick:() => alert('test')}]} />
    </div>
```

I could not find a good way to determine the width of the menu items to get them to the width of the parent (mostly, because its not the DOM parent)

`popoverProps` allows to send some properties to the popover, like the anchor client rect, but I didn't see something, which could account for width or min-width

Reply:
@websupporter Hmm, OK, I guess. I know the WP components dropdown isn't great due to the `controls` parameter, but on the other hand, it handles keyboard navigation (also arrows, etc.), it auto-closes on click (we should add that too, so people don't double-click and create multiple workflows), etc.

This seems to work quite OK:
```ts
<style>{`
  .mailpoet-option-button-opener {
    background: var(--wp-admin-theme-color);
    color: white;
  }
  .mailpoet-option-button-opener svg {
    fill: white;
  }
  .mailpoet-option-button-opener .is-opened svg {
    transform: scale(-1, -1);
    transform-origin: center 12.5px;
  }
`}</style>
<DropdownMenu
  className="mailpoet-option-button-opener"
  label={__('More', 'mailpoet')}
  icon={chevronDown}
  controls={[{ title: 'New workflow', icon: undefined }]}
  popoverProps={{ position: 'bottom left' }}
/>
```

Result:

<img width="235" alt="Screenshot 2022-11-02 at 14 28 48" src="https://user-images.githubusercontent.com/141436/199502392-49a09fde-b5a6-41fd-a483-f7a2e1afe877.png">

And the style of the menu below can be CSSed a bit as well. As for the width, I'd omit that, maybe. It's not practical for translations anyway — the button will have different widths in different languages, the dropdown menu items too. Anyway, it's one option, I guess.

Either this or at least adding auto-close on item click.